providers/implementations/keymgmt/ecx_kmgmt.c

```diff
@@ -706,15 +707,17 @@ static int ecx_key_pairwise_check(const ECX_KEY *ecx, int type)
 static int ecx_validate(const void *keydata, int selection, int type, size_t keylen)
 {
     const ECX_KEY *ecx = keydata;
-    int ok = 0;
+    int ok = keylen == ecx->keylen;
 
     if (!ossl_prov_is_running())
         return 0;
 
-    assert(keylen == ecx->keylen);
+    if ((selection & ECX_POSSIBLE_SELECTIONS) == 0)
```

Comment:
Should this happen after the `if(!ok)` check below?

Reply:
I see the keylen == ecx->keylen check as part of the validation checks and then it should be done after this short-cut out. But of course it can be seen both ways.